apis/Google.Cloud.Firestore.Data/Google.Cloud.Firestore.Data/FirestoreDb.cs

```diff
@@ -0,0 +1,148 @@
+        }
+
+        /// <summary>
+        /// Creates an instance for the specified project and database, using the specified <see cref="FirestoreClient"/>
+        /// for RPC operations.
+        /// </summary>
+        /// <param name="projectId">The ID of the Google Cloud Platform project that contains the database. May be null, in which case
+        /// the project will be automatically detected if possible.</param>
+        /// <param name="databaseId">The ID of the database within the project. May be null, in which case the default database will be used.</param>
+        /// <param name="client">The client to use for RPC operations. May be null, in which case a client will be created with default credentials.</param>
+        /// <returns>A new instance.</returns>
+        public static FirestoreDb Create(string projectId = null, string databaseId = null, FirestoreClient client = null) =>
+            new FirestoreDb(
+                projectId ?? Platform.Instance().ProjectId,
+                databaseId ?? DefaultDatabaseId,
+                client ?? FirestoreClient.Create());
+
+        /// <summary>
+        /// Asynchronously creates an instance for the specified project and database, using the specified <see cref="FirestoreClient"/>
+        /// for RPC operations.
+        /// </summary>
+        /// <param name="projectId">The ID of the Google Cloud Platform project that contains the database. May be null, in which case
+        /// the project will be automatically detected if possible.</param>
+        /// <param name="databaseId">The ID of the database within the project. May be null, in which case the default database will be used.</param>
+        /// <param name="client">The client to use for RPC operations. May be null, in which case a client will be created with default credentials.</param>
+        /// <returns>A new instance.</returns>
+        public static async Task<FirestoreDb> CreateAsync(string projectId = null, string databaseId = null, FirestoreClient client = null) =>
+            new FirestoreDb(
+                projectId ?? (await Platform.InstanceAsync().ConfigureAwait(false)).ProjectId,
+                databaseId ?? DefaultDatabaseId,
+                client ?? await FirestoreClient.CreateAsync().ConfigureAwait(false)
+            );
+
+        // TODO: Rename Collection and Document to GetCollection, GetDocument or GetCollectionReference or GetDocumentReference?
+        // (Collection/Document follow the suggested convention for all languages. They feel odd to start with, but increasingly pleasant.)
+
+        /// <summary>
+        /// Creates a local <see cref="CollectionReference"/> for the given path, which must include
+        /// an odd number of slash-separated identifiers. This does not perform any remote operations.
+        /// </summary>
+        /// <param name="path">The collection path, e.g. <c>col1/doc1/col2</c>.</param>
+        /// <returns>A document reference.</returns>
```

Comment:
@evildour You missed this! Can't believe I got something past you ;) (Fixed)

Reply:
Haha. Well I haven't have my minimum 4 cups of coffee before starting work this morning, so that might explain it :)